09-writing-advanced-plans/README.md

```diff
@@ -46,20 +46,26 @@ In the previous exercise you ran tasks and commands within the context of a plan
 3. Create a plan and save it as `modules/exercise9/plans/yesorno.pp`:
 
     ```puppet
+    # TargetSpec accepts a comma-separated list of nodes
     plan exercise9::yesorno (TargetSpec $nodes) {
+      # Run the 'exercise9::yesorno' task on the nodes you specify.
       $results = run_task('exercise9::yesorno', $nodes)
-      $subset = $results.filter |$result| { $result[answer] == true }.map |$result| { $result.target }
-      run_command("uptime", $subset)
+
+      # Puppet uses immutable variables. That means we have to operate on data, in
+      # this case a ResultSet containing a list of Results. Those are documented in
+      # Bolt's docs, but effectively its a list of result data parsed from JSON
+      # objects returned by the task. Collect - "filter" - only the Result objects
+      # where the task printed '{"answer": true}'.
+      $answered_true = $results.filter |$result| { $result[answer] == true }
+
+      # Result objects also include a reference to the target they came from. Get a
+      # list of the targets that answered 'true'.
+      $nodes_subset = $answered_true.map |$result| { $result.target }
+
+      # Run the 'uptime' command on the list of targets that answered 'true'.
+      run_command('uptime', $nodes_subset)
     }
     ```
-    
-    This plan: 
```

Comment:
This might be a deviation from general practice elsewhere in the doc. It's probably better to provide inline description rather than do it after-the-fact (where you have to identify what line points to what part of the code).

Reply:
I don't think they would, which is why the links come after the example.